alexandra/app.py

```diff
@@ -62,14 +61,28 @@ def dispatch_request(self, body):
         elif req_type == 'IntentRequest':
             intent = body['request']['intent']['name']
             intent_fn = self.intent_map.get(intent, self.unknown_intent_fn)
+            slots = body['request']['intent'].get('slots', {})
+
+            # Python 2.7 compatibility.
+            try:
+                slot_list = slots.iteritems()
+            except AttributeError:
+                slot_list = slots.items()
 
             slots = {
                 slot['name']: slot.get('value')
-                for _, slot in
-                body['request']['intent'].get('slots', {}).iteritems()
+                for _, slot in slot_list
             }
 
-            if intent_fn.func_code.co_argcount == 2:
+            code_list = inspect.getmembers(intent_fn, inspect.iscode)
+
+            # Python 2.7 compatibility.
+            if len(code_list) == 1:
+                (_, code) = code_list[0]
+            else:
+                (_, code) = code_list[1]
```

Comment:
https://docs.python.org/3/library/inspect.html#inspect.getargspec Deprecated since 3.0 and signature doesn't exist in 2, so I preferred this over getargspec. I'd rather do this than use deprecated functions.